ckanext/switzerland/plugin.py

```diff
@@ -390,6 +390,27 @@ def before_view(self, pkg_dict):
         if not self.is_supported_package_type(pkg_dict):
             return pkg_dict
 
+        # create resource views if necessary
+        user = logic.get_action('get_site_user')({'ignore_auth': True}, {})
+        context = {
+            'model': model,
+            'session': model.Session,
+            'user': user['name']
+        }
+        logic.check_access('package_create_default_resource_views', context)
+
+        # get the dataset via API, as the pkg_dict does not contain all fields
+        dataset = logic.get_action('package_show')(
+            context,
+            {'id': pkg_dict['id']}
+        )
+
+        # Make sure resource views are created before showing a dataset
```

Comment:
Subsequently [add_views_to_resource](https://github.com/ckan/ckan/blob/d73326ad4beb3743cb4a347b15c04dace8c8654b/ckan/lib/datapreview.py#L202) is called. It creates a view for every default type that a) has not yet been created and b) claims it can preview this type of data

So once the views have been created, tnis method will simply detect that they already exist and return.